src/pull_request.ts

```diff
@@ -80,26 +112,52 @@ export async function handlePullRequest() {
   });
   info(`generated pull request summary: ${summary.title}`);
 
-  // Update PR title and description
-  //   await octokit.rest.pulls.update({
-  //     ...context.repo,
-  //     pull_number: pull_request.number,
-  //     title: summary.title,
-  //     body: summary.description,
-  //   });
-  //   info(`updated pull request title and description`);
+  // Update PR title if @presubmitai is mentioned in the title
+  if (pull_request.title.includes("@presubmitai")) {
+    info(`title contains @presubmitai, so generating a new title`);
+    await octokit.rest.pulls.update({
+      ...context.repo,
+      pull_number: pull_request.number,
+      title: summary.title,
+      // body: summary.description,
+    });
+  }
 
   // Update overview comment with the walkthrough
   await octokit.rest.issues.updateComment({
     ...context.repo,
     comment_id: overviewComment.id,
-    body: buildWalkthroughMessage(summary),
+    body: buildWalkthroughMessage(
+      summary,
+      commits.map((c) => c.sha)
+    ),
   });
   info(`updated overview comment with walkthrough`);
 
-  // Review PR code changes
+  // ======= START REVIEW =======
+
+  // Check if there are any incremental changes
+  const lastCommitReviewed = commitsReviewed.length
+    ? commitsReviewed[commitsReviewed.length - 1]
+    : null;
+  const incrementalDiff =
+    lastCommitReviewed && lastCommitReviewed != pull_request.head.sha
+      ? await octokit.rest.repos.compareCommits({
+          ...context.repo,
+          base: lastCommitReviewed,
+          head: pull_request.head.sha,
+        })
+      : null;
+  let filesToReview = fileDiffs;
+  if (incrementalDiff?.data?.files) {
+    // If incremental review, only consider files that were modified within incremental change.
+    filesToReview = filesToReview.filter((f) =>
+      incrementalDiff.data.files?.some((f2) => f2.filename === f.filename)
+    );
+  }
```

Comment:
The optional chaining on `incrementalDiff.data.files` in the `some()` method could be moved outside the filter to avoid potential undefined access. Consider restructuring as:
```typescript
const incrementalFiles = incrementalDiff?.data?.files;
if (incrementalFiles) {
  filesToReview = filesToReview.filter((f) =>
    incrementalFiles.some((f2) => f2.filename === f.filename)
  );
}
```